pkg/proxy/healthcheck/healthcheck_test.go

```diff
@@ -438,7 +438,7 @@ func TestServerWithSelectiveListeningAddress(t *testing.T) {
 
 	// limiting addresses to loop back. We don't want any cleverness here around getting IP for
 	// machine nor testing ipv6 || ipv4. using loop back guarantees the test will work on any machine
-	nodePortAddresses := utilproxy.NewNodePortAddresses(v1.IPv4Protocol, []string{"127.0.0.0/8"})
+	nodePortAddresses := utilproxy.NewNodePortAddresses(v1.IPv4Protocol, []string{"127.0.0.0/8"}, true)
```

Comment:
Within the context of this PR the constructor keeps changing, but that's just because I was fixing things one by one. There's no particular reason to think it will change again after this...

Reply:
(Though the ambiguous untyped boolean arg does suck...)

Reply:
yeah, but if we want to expose it who knows ... maybe mulitnetwork or some new development bring new options.
I

Reply:
Maybe any feature would bring new options to any API.

My point is that you are suggesting we use an "options" pattern because you saw that the API kept getting added to. But the reason that that happened is just because I intentionally split the introduction of the full API over multiple commits for ease of review. If I had just introduced the API with all of the arguments already there, you presumably wouldn't have said anything.

I'm not saying the API won't ever change again, I'm just saying that the likelihood it will change again is exactly the same as for every other API we have, and since we don't _normally_ use options patterns on constructors, there's no reason to do so here either.